src/main/java/org/usfirst/frc4904/robot/RobotMap.java

```diff
@@ -1,32 +1,91 @@
 
 public class RobotMap {
     public static class Port {
         public static class HumanInput {
+            public static final int joystick = 0;
             public static final int xboxController = 1;
+        }
 
+        public static class CANMotor {
+            public static final int elevatorMotorA = -1; // not final numbers
+            public static final int elevatorMotorB = -1;
         }
-        public static class CANMotor {}
-        public static class PWM {}
-        public static class CAN {}
-        public static class Pneumatics {}
+
+        public static class CAN {
+            public static final int elevatorEncoder = 1; // also not final #
+        }
+
+        public static class Pneumatics {
+            public static final PCMPort fourBarLever = new PCMPort(-1, -1, -1);
+        }
+
+        public static class Digital {
+            public static final int elevatorSwitchBottomPort = -1; // TODO: Not final values.
+            public static final int elevatorSwitchTopPort = -1;
+        }
+    }
+
+    public static class Metrics {
+        public static final double ELEVATOR_INNER_HEIGHT = -1; // TODO: set to actual value
     }
-    
-    public static class Metrics{}
+
+    public static class PID {
+        public static class Elevator {
+            public static final double P = 1; // TODO: TUNE
+            public static final double I = 1;
+            public static final double D = 1;
+            public static final double F = 1;
+        }
+    }
+
     public static class Component {
         public static CustomXbox driverXbox;
+        public static CANEncoder elevatorEncoder;
+        public static FourBarLinkage fourBar;
     }
+
     public static class HumanInput {
         public static class Driver {
             public static CustomXbox xbox;
         }
-        public static class Operator {}
+
+        public static class Operator {
+            public static CustomJoystick joystick;
+        }
     }
+
+    public static class Input {
+        public static CustomDigitalLimitSwitch elevatorSwitchBottom;
+        public static CustomDigitalLimitSwitch elevatorSwitchTop;
+    }
+
     public RobotMap() {
         Component.driverXbox = new CustomXbox(Port.HumanInput.xboxController);
-		Component.driverXbox.setDeadZone(0.1);
+        Component.driverXbox.setDeadZone(0.1);
+        HumanInput.Operator.joystick = new CustomJoystick(Port.HumanInput.joystick);
+        Component.elevatorEncoder = new CANEncoder(Port.CAN.elevatorEncoder);
+        Component.elevatorEncoder.setDistancePerPulse(FourBarLinkage.TICK_MULTIPLIER);
+        Component.fourBar = new FourBarLinkage(new SolenoidSubsystem("FourBarLinkage", SolenoidState.RETRACT,
+            Port.Pneumatics.fourBarLever.buildDoubleSolenoid()),
+            new PositionSensorMotor("Elevator",
+                new CustomPIDController(PID.Elevator.P, PID.Elevator.I, PID.Elevator.D, PID.Elevator.F,
```

Comment:
Need to construct the PID controller outside of this constructor (store it in RobotMap) in order to set its tolerances and such

Reply:
And for sending the values over to the driverstation for elevator tuning